### PR TITLE
add more common cfg to DefaultOptions

### PR DESCRIPTION
**Pull request by weberlo**

### CODE REVIEW

File: python/tvm/micro/build.py [hunk 1]
```diff
@@ -91,15 +91,26 @@ def _generate_mod_wrapper(src_path):
 
 _CRT_DEFAULT_OPTIONS = {
   'ccflags': ['-std=c++11'],
+  'ldflags': ['-std=gnu++14'],
   'include_dirs': [f'{TVM_ROOT_DIR}/include',
                    f'{TVM_ROOT_DIR}/3rdparty/dlpack/include',
                    f'{TVM_ROOT_DIR}/3rdparty/mbed-os/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_11/libraries/crc16/',
-                   f'{TVM_ROOT_DIR}/3rdparty/dmlc-core/include'],
+                   f'{TVM_ROOT_DIR}/3rdparty/dmlc-core/include',
+                   f'{CRT_ROOT_DIR}/include'
```
Comment (areusch) [inline python/tvm/micro/build.py:99]: this one's already in there

File: tests/python/unittest/test_crt.py [hunk 2]
```diff
@@ -69,20 +70,11 @@ def _make_cortex_m33_sess(mod):
...
-  lib_opts = copy.deepcopy(bin_opts)
-  lib_opts['profile']['common'].append('-Werror')
-  lib_opts['cflags'] = ['-Wno-error=incompatible-pointer-types']
+  opts = tvm.micro.DefaultOptions(f'{project_dir}/crt')
 
   if BUILD:
-    micro_binary = tvm.micro.build_static_runtime(workspace, compiler, mod, lib_opts, bin_opts)
-  lib_opts['cflags'].pop()
+    micro_binary = tvm.micro.build_static_runtime(
+      workspace, compiler, mod, lib_opts=opts['lib_opts'], bin_opts=opts['bin_opts'])
```
Comment (areusch) [inline tests/python/unittest/test_crt.py]: maybe just **opts?

File: tests/python/unittest/test_crt.py [hunk 3]
```diff
@@ -69,20 +70,11 @@ def _make_cortex_m33_sess(mod):
...
-  bin_opts.setdefault('profile', {})['common'] = ['-Wno-unused-variable']
-  bin_opts.setdefault('ldflags', []).append('-std=gnu++14')
-  bin_opts.setdefault('include_dirs', []).append(f'{project_dir}/crt')
-  crt_include_dir = os.path.realpath(os.path.join(tvm.micro.CRT_ROOT_DIR, 'include'))
-  bin_opts.setdefault('include_dirs', []).append(crt_include_dir)
-
-  lib_opts = copy.deepcopy(bin_opts)
-  lib_opts['profile']['common'].append('-Werror')
-  lib_opts['cflags'] = ['-Wno-error=incompatible-pointer-types']
+  opts = tvm.micro.DefaultOptions(f'{project_dir}/crt')
```
Comment (areusch) [inline tests/python/unittest/test_crt.py:73]: can you also update whatever is in tests/micro?

File: python/tvm/micro/build.py (hunk 1 above)
Comment (weberlo) [reply python/tvm/micro/build.py:99]: the first one uses `TVM_ROOT_DIR` as the prefix and the last one is `CRT_ROOT_DIR`

File: tests/python/unittest/test_crt.py (hunk 2 above)
Comment (weberlo) [reply tests/python/unittest/test_crt.py]: yeah. that's way cleaner

File: python/tvm/micro/build.py (hunk 1 above)
Comment (areusch) [reply python/tvm/micro/build.py:99]: oh good point

**Pull request merged**